common/source/docs/common-brushless-escs.rst

```diff
@@ -36,18 +47,97 @@ The DShot ESC protocol's key advantages are:
+
+If the craft has been setup for DShot commands then ArduPilot will supply the correct command at startup in order to set the ESCs in reversible mode.
+
+In a similar fashion, normal output rotation direction can be reversed by setting :ref:`SERVO_BLH_RVMASK<SERVO_BLH_RVMASK>` without any changes needing to be made through ESC setup software (e.g. BLHeliSuite). This can also be used on ESCs with forward and reversed active operation, ie reversible ESCs, to set the "forward" direction's rotation.
+
+.. note:: Currently, ArduPilot only supports the use of reversible ESCs for Plane and Rover, not Copter.
+
+.. _output-groups:
+
+Autopilot Output Capabilities
+=============================
+
+While all the servo/motor outputs of an ArduPilot autopilot are capable of Normal PWM operation at 50Hz frame rates, not all are capable of other ESC protocol configurations. And, usually, these configurations must apply to pre-designated groups of outputs, even if they are not all driving an ESC. So the following cautions apply:
+
+#. The 8 "MAIN" outputs of autopilots using an IOMCU (like PixHawk and Cube), cannot be used for  protocols other than Normal PWM and OneShot. On these autopilots, only the additional "AUX" outputs can properly support OneShot125 and DShot protocols. If you attempt to set a "MAIN" output to DShot, then normal PWM output will occur, even though it has been set to a DShot protocol.
+
+#. For :ref:`Pixracer <common-pixracer-overview>` and :ref:`other boards <common-autopilots>` without a separate IOMCU coprocessor, all servo/motor outputs can be used.
+
+#. Groups of outputs sharing a common timer, MUST have the same advanced configuration. Usually, these are specified in the autopilot's hardware description linked from the :ref:`common-autopilots` page. For example, if an output is configured for DShot in a group, then you cannot use another output in that group for Normal PWM ESC **or** normal PWM servo operation.
+
+.. note:: Everytime the autopilot initializes, it sends a log message to the ground control station, showing which outputs are PWM/Oneshot/or DShot. The remaining higher numbered outputs are assigned as GPIOs.
+
+.. image:: ../../../images/RCOutbanner.jpg
+
+
+.. _esc-telemetry:
+
+Telemetry
+=========
+
+Many brushless ESCs offer telemetry reporting of important ESC data, such as RPM, Temperature, Current, etc. for use in OSDs or for controlling ArudPilot functions like variable center frequency noise filters. This can be via a serial connection to one of the autopilot's UART RX inputs, whose ``SERIALx_PROTOCOL`` has been set to "16" (ESC Telemetry), or directly over its control connection if it has Bi-Directional DShot capability (only available on certain BLHeli32 capable ESCs at present).
```

Comment:
ESC telemetry is a BLHeli only thing right?  Normal DShot doesn't provided feedback I think so this should go up in the BLHeli section.

Reply:
no, AFAIK you can have ESCs that are not BLHeli capable, that have ESC telemetry via a separate wire vs over the digital control link (which is a BLHeli feature and requires digital communication of control and telem)...at some back as this evolved,  I remember seeing an ESC that did not advertise BLHeli capablity and was one-shot125 and had ESC telem

Reply:
but I cant find one today....I will move all Telemetry to the BLHeli page